optuna/terminator/improvement/evaluator.py

```diff
@@ -128,3 +128,58 @@ def _validate_input(
                 "The intersection search space is empty. This condition is not supported by "
                 f"{cls.__name__}."
             )
+
+
+@experimental_class("3.4.0")
+class BestValueStagnationEvaluator(BaseImprovementEvaluator):
+    """Evaluates the stagnation period of the best value in an optimization process.
+
+    This class is initialized with a maximum stagnation period (`max_stagnation_trials`)
+    and is designed to evaluate the remaining trials before reaching this maximum period
+    of allowed stagnation.
```

Comment:
I think it should be clearly documented that this evaluator is to be used with `StaticErrorEvaluator(0)`.

Reply:
Maybe it would be kind to add an example?

Reply:
@contramundum53 That sounds reasonable. However, those appear to be optional items, so we can address them in follow-up PRs.